contracts-periphery/snapshot/.gas

```diff
@@ -1,99 +1,199 @@
-│ sendEth                      ┆ 37476           ┆ 37547  ┆ 37476  ┆ 39476  ┆ 2684    │
-├╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌┤
-│ sendToken                    ┆ 24180           ┆ 24625  ┆ 24180  ┆ 48261  ┆ 2688    │
-├╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌┤
-│ toll                         ┆ 321             ┆ 1334   ┆ 2321   ┆ 2321   ┆ 300     │
-├╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌┤
-│ withdrawTokenAndCall         ┆ 195513          ┆ 195513 ┆ 195513 ┆ 195513 ┆ 2       │
-├╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌┤
-│ withdrawTokenAndCallOnBehalf ┆ 208162          ┆ 208162 ┆ 208162 ┆ 208162 ┆ 2       │
-╰──────────────────────────────┴─────────────────┴────────┴────────┴────────┴─────────╯
-╭────────────────────────┬─────────────────┬───────┬────────┬───────┬─────────╮
-│ UniswapV3Pool contract ┆                 ┆       ┆        ┆       ┆         │
-╞════════════════════════╪═════════════════╪═══════╪════════╪═══════╪═════════╡
-│ Deployment Cost        ┆ Deployment Size ┆       ┆        ┆       ┆         │
-├╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌┤
-│ 0                      ┆ 0               ┆       ┆        ┆       ┆         │
-├╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌┤
-│ Function Name          ┆ min             ┆ avg   ┆ median ┆ max   ┆ # calls │
-├╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌┤
-│ swap                   ┆ 65822           ┆ 73481 ┆ 73481  ┆ 81141 ┆ 8       │
-╰────────────────────────┴─────────────────┴───────┴────────┴───────┴─────────╯
-╭─────────────────┬─────────────────┬───────┬────────┬───────┬─────────╮
-│ WETH9 contract  ┆                 ┆       ┆        ┆       ┆         │
-╞═════════════════╪═════════════════╪═══════╪════════╪═══════╪═════════╡
-│ Deployment Cost ┆ Deployment Size ┆       ┆        ┆       ┆         │
-├╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌┤
-│ 0               ┆ 0               ┆       ┆        ┆       ┆         │
-├╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌┤
-│ Function Name   ┆ min             ┆ avg   ┆ median ┆ max   ┆ # calls │
-├╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌┤
-│ balanceOf       ┆ 534             ┆ 534   ┆ 534    ┆ 534   ┆ 4       │
-├╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌┤
-│ transfer        ┆ 29962           ┆ 29962 ┆ 29962  ┆ 29962 ┆ 8       │
-├╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌┤
-│ withdraw        ┆ 7379            ┆ 7379  ┆ 7379   ┆ 7379  ┆ 4       │
-╰─────────────────┴─────────────────┴───────┴────────┴───────┴─────────╯
+Compiling 3 files with 0.8.13
```

Comment:
I'm a bit confused by this file. How does it relate to `.gas-snapshot`? Also, when I run `make snapshot-gas` on this branch, I get a big diff in this file. Is that supposed to happen? Maybe some updates to the README to explain how gas snapshotting works would be helpful.

Reply:
Context from @mds1:

```
* `forge snapshot` runs your tests and reports gas usage of each test, and creates a `.gas-snapshot` file at the root
* `forge test --gas-report` outputs a table of specific functions and their min/max/med/avg gas usage, but it just prints to the terminal so you need to manually dump it to a file
```

Reply:
Since we're including both files, we want our `snapshot-gas` command to update both files